From 253021cafc14af9c70b8380e91b5c3796d655125 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Sun, 5 Jun 2016 12:57:38 +0300
Subject: [PATCH] dd-list: make the file argument of the '-s' option mandatory

As discussed in https://rt.cpan.org/Public/Bug/Display.html?id=39052
dd-list was relying on buggy Getopt::Long behaviour to parse 'dd-list -s
<sources>': GNU getopt compatibility would require 'dd-list -s=<sources>'
for optional arguments.

There doesn't seem to be a use case for specifying '-s' without a
source file, so fix the problem by making it mandatory.
---
 scripts/dd-list.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/dd-list.pl b/scripts/dd-list.pl
index 71f277c..f7c4b50 100755
--- a/scripts/dd-list.pl
+++ b/scripts/dd-list.pl
@@ -100,7 +100,7 @@ GetOptions(
     "help|h" => sub { help(); exit },
     "stdin|i" => \$use_stdin,
     "dctrl|d" => \$use_dctrl,
-    "sources|s:s@" => \$source_files,
+    "sources|s=s@" => \$source_files,
     "uploaders|u!" => \$show_uploaders,
     'z|uncompress' => \$opt_uncompress,
     "print-binary|b" => \$print_binary,
-- 
2.8.1

